dashboard/app/models/pd/workshop_facilitator_daily_survey.rb

```diff
@@ -89,5 +89,14 @@ def self.create_placeholder!(user_id:, day:, pd_session_id:, facilitator_id:, fo
         submission_id: submission_id,
       )
     end
+
+    # @override
+    def duplicate?
+      # See if this user already has a submission for this workshop, session, facilitator, & form.
+      # Note: this duplicate record would fail the uniqueness validation
+      self.class.where(
```

Comment:
Prefer `self.class.exists?(...)`
Also why use self.class here? Does this have a subclass?

Reply:
As far as `self.class`, referring to the class by name inside its own definition seems weird to me so I prefer this way, but I'm happy to switch to the class name if you feel strongly about it.